azurerm/internal/services/web/app_service_certificate_binding_resource.go

```diff
@@ -0,0 +1,223 @@
+	}
+
+	certDetails, err := certClient.Get(ctx, id.CertificateId.ResourceGroup, id.CertificateId.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(certDetails.Response) {
+			return fmt.Errorf("retrieving App Service Certificate %q (Resource Group %q), not found", id.CertificateId.Name, id.CertificateId.ResourceGroup)
+		}
+		return fmt.Errorf("failed reading App Service Certificate %q (Resource Group %q): %+v", id.CertificateId.Name, id.CertificateId.ResourceGroup, err)
+	}
+
+	if certDetails.Thumbprint == nil {
+		return fmt.Errorf("could not read thumbprint from certificate %q (resource group %q): %+v", id.CertificateId.Name, id.CertificateId.ResourceGroup, err)
+	}
+	thumbprint := certDetails.Thumbprint
+
+	binding, err := client.GetHostNameBinding(ctx, id.HostnameBindingId.ResourceGroup, id.HostnameBindingId.SiteName, id.HostnameBindingId.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(binding.Response) {
+			return fmt.Errorf("retrieving Custom Hostname Binding %q (App Service %q / Resource Group %q): %+v", id.HostnameBindingId.Name, id.HostnameBindingId.SiteName, id.HostnameBindingId.ResourceGroup, err)
+		}
+		return fmt.Errorf("retrieving Custom Hostname Certificate Binding %q with certificate name %q (App Service %q / Resource Group %q): %+v", id.HostnameBindingId.Name, id.HostnameBindingId.SiteName, id.HostnameBindingId.SiteName, id.HostnameBindingId.ResourceGroup, err)
+	}
+
+	props := binding.HostNameBindingProperties
+	if props != nil {
+		if props.Thumbprint != nil && *props.Thumbprint == *thumbprint {
+			return tf.ImportAsExistsError(appServiceHostnameBindingResourceName, id.ID(""))
+		}
+	}
+
+	locks.ByName(id.HostnameBindingId.SiteName, appServiceHostnameBindingResourceName)
+	defer locks.UnlockByName(id.HostnameBindingId.SiteName, appServiceHostnameBindingResourceName)
+
+	binding.HostNameBindingProperties.SslState = web.SslState(d.Get("ssl_state").(string))
+	binding.HostNameBindingProperties.Thumbprint = thumbprint
+
+	if _, err := client.CreateOrUpdateHostNameBinding(ctx, id.HostnameBindingId.ResourceGroup, id.HostnameBindingId.SiteName, id.HostnameBindingId.Name, binding); err != nil {
+		return fmt.Errorf("creating/updating Custom Hostname Certificate Binding %q with certificate name %q (App Service %q / Resource Group %q): %+v", id.HostnameBindingId.Name, id.CertificateId.Name, id.HostnameBindingId.SiteName, id.HostnameBindingId.ResourceGroup, err)
+	}
+
+	d.SetId(id.ID(""))
+
+	return resourceArmAppServiceCertificateBindingRead(d, meta)
+}
+
+func resourceArmAppServiceCertificateBindingRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CertificateBindingID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("hostname_binding_id", id.HostnameBindingId.ID(""))
+	d.Set("certificate_id", id.CertificateId.ID(""))
```

Comment:
tbh we should probably move this below the Get below, since otherwise this is disregarded if it's not found